fix bugs in _prepare_text and IntruderScorer prompt formatting

d0rbu opened the pull request:
in `delphi.scorers.classifier.sample._prepare_text`, i believe i've found a bug when calling it with `n_incorrect > 1`(i.e. producing a false positive sample by incorrectly highlighting more than 1 non-activating token). i've added a comment to the location of the bug with an explanation.

additionally, in `delphi.scorers.classifier.intruder.IntruderScorer._build_prompt` (the method used to format a list of intruder examples into a prompt for the scorer model), there is an inconsistency with few-shot examples that i've also highlighted in a comment

there are also a bunch of smaller stylistic/readability changes, feel free to push back on these.

d0rbu commented on delphi/scorers/classifier/intruder.py:
```diff
@@ -275,7 +269,7 @@ def _build_prompt(
         """
 
         examples = "\n".join(
-            f"Example {i}: {example}" for i, example in enumerate(sample.examples)
```
in `delphi.scorers.classifier.prompts.intruder_prompt`, there are some samples for a few-shot prompt that do not add a space after the "Example {i}:". for example, `DSCORER_EXAMPLE_FOUR` in that file contains the following sample:

```
Example 0: You<< guys>> support me in many other ways already and
Example 1: birth control access but I assure that << guys>> in Kentucky
Example 2: gig! I hope you <<guys>> LOVE her, and please be nice,
Example 3:American, told Hannity that you<< guys>> are playing the race card.
```

most of the examples happen to have a space after the "Example {i}:" because the first token in the sequence starts with a space, but the last example shows that this is not necessarily the case. i've checked this by checking out the original activating examples on neuronpedia and confirming that those examples do indeed start with spaces.

SrGonao reviewed (commented):
I would rather not have all these different changes in the same pull request, but we can do it like this. I left some comments on things I didnt quite understand but most of it seems fine. Will wait on your updates. Reach me on discord if you want to chat more about delphi

d0rbu commented on delphi/scorers/classifier/intruder.py:
```diff
@@ -153,38 +157,37 @@ def _prepare_and_batch(self, record: LatentRecord) -> list[IntruderSentence]:
-            majority_examples = []
-            active_tokens = 0
+            # highlights the active tokens with <<>> markers
+            examples = []
```
hm, my issue is that it does not *only* contain the activating majority samples; later on, after `majority_examples.insert(intruder_index, intruder_sentence)`, it also includes the intruder.

i think a better solution may be to keep the name `majority_samples` up until that point and reassigning it to a new `examples` variable that contains both the majority and intruder samples

SrGonao replied:
Agreed

d0rbu commented:
> I would rather not have all these different changes in the same pull request

very reasonable! i've taken them out for now, i can just put them in a separate PR. i've addressed the comments anyway to explain the changes

d0rbu commented on delphi/pipeline.py:
```diff
@@ -161,8 +161,9 @@ async def process_item(self, item: Any, semaphore: asyncio.Semaphore) -> Any:
-                else:
-                    pass
+                if result is None:
+                    break
```
once a pipe returns `None`, we iterate through the rest of the pipes and simply hit the `pass` until we return the result, which is None. this skips that pointless extra iteration; perhaps it may be clearer if we replace `break` with `return result` or `return None`?
